web/src/features/moreCast2/components/MoreCast2DataGrid.tsx

```diff
@@ -25,90 +31,77 @@ const useStyles = makeStyles({
-    }
-    // Check if the user has edited the value. If so, update the value and choice to reflect the Manual edit.
-    if (newValue.toFixed(precision) !== params.row[field].value.toFixed(precision)) {
-      params.row[field].choice = ModelChoice.MANUAL
-      params.row[field].value = newValue
-    }
-
-    return { ...params.row }
+  const handleClose = () => {
+    setContextMenu(null)
   }
 
-  const gridColumnDefGenerator = (field: string, headerName: string, precision: number) => {
-    return {
-      field: field,
-      disableColumnMenu: true,
-      disableReorder: true,
-      editable: true,
-      headerName: headerName,
-      sortable: false,
-      type: 'number',
-      width: 120,
-      valueFormatter: (params: GridValueFormatterParams) => predictionItemValueFormatter(params, precision),
-      valueGetter: (params: GridValueGetterParams) => predictionItemValueGetter(params, precision),
-      valueSetter: (params: GridValueSetterParams) => predictionItemValueSetter(params, field, precision)
-    }
-  }
+  const loading = useSelector(selectMorecast2TableLoading)
 
-  const columns: GridColDef[] = [
-    { field: 'stationName', flex: 1, headerName: 'Station', maxWidth: 200 },
-    {
-      field: 'forDate',
-      disableColumnMenu: true,
-      disableReorder: true,
-      flex: 1,
-      headerName: 'Date',
-      maxWidth: 250,
-      sortable: false,
-      valueFormatter: (params: GridValueFormatterParams<DateTime>) => {
-        return params.value.toLocaleString(DateTime.DATE_MED)
-      }
-    },
-    gridColumnDefGenerator('temp', 'Temp', 1),
-    gridColumnDefGenerator('rh', 'RH', 0),
-    gridColumnDefGenerator('windDirection', 'Wind Dir', 0),
-    gridColumnDefGenerator('windSpeed', 'Wind Speed', 1),
-    gridColumnDefGenerator('precip', 'Precip', 1)
-  ]
+  const columns: GridColDef[] = MORECAST2_FIELDS.map(field => field.generateColDef())
 
   return (
     <div className={classes.root} data-testid={`morecast2-data-grid`}>
       <DataGrid
         components={{
           LoadingOverlay: LinearProgress
         }}
+        onColumnHeaderClick={handleColumnHeaderClick}
         loading={loading}
         columns={columns}
         rows={rows}
       ></DataGrid>
+      <Menu
+        open={contextMenu !== null}
+        onClose={handleClose}
+        anchorReference="anchorPosition"
+        anchorPosition={contextMenu !== null ? { top: contextMenu.mouseY, left: contextMenu.mouseX } : undefined}
+        slotProps={{
+          root: {
+            onContextMenu: e => {
+              e.preventDefault()
+              handleClose()
+            }
+          }
+        }}
+      >
+        <MenuItem
+          disableRipple
+          sx={{
+            '&:hover': {
+              backgroundColor: 'transparent' // remove the background color on hover
+            },
+            '&.Mui-selected': {
+              backgroundColor: 'transparent' // remove the background color when selected
+            }
```

Comment:
Trivial, but we get some odd highlighting when the input has focus and the cursor is outside the menu.

![image](https://user-images.githubusercontent.com/7094320/225164684-60e439da-df09-46af-b3c7-7b6961f65094.png)


Adding this should fix it:
```
            '&.Mui-focusVisible': {
              backgroundColor: 'transparent'
            }
```